src/main/scala/com/fulcrumgenomics/bam/SplitBamByReadGroup.scala

```diff
@@ -0,0 +1,113 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package com.fulcrumgenomics.bam
+
+import com.fulcrumgenomics.cmdline.{ClpGroups, FgBioTool}
+import com.fulcrumgenomics.commons.CommonsDef.{PathPrefix, PathToBam, SafelyClosable, javaIterableToIterator, javaIteratorAsScalaIterator}
+import com.fulcrumgenomics.commons.io.{Io, PathUtil}
+import com.fulcrumgenomics.commons.util.{LazyLogging, SimpleCounter}
+import com.fulcrumgenomics.sopt.{arg, clp}
+import com.fulcrumgenomics.util.ProgressLogger
+import htsjdk.samtools._
+
+@clp(group = ClpGroups.SamOrBam, description=
+  """
+    |Splits a BAM into multiple BAMs per-read group (or library).
+    |
+    |The resulting BAM will be named `<output-prefix>.<read-group-id>.bam`, or `<output-prefix>.<library-name>.bam` when
+    |splitting by the library.  All reads without a read group, or without a library when splitting by library, will be
+    |written to `<output-prefix>.unknown.bam`.
+  """)
+class SplitBamByReadGroup
+( @arg(flag='i', doc="Input SAM or BAM file.") val input: PathToBam,
+  @arg(flag='o', doc="Output prefix for all SAM or BAM files (ex. output/sample-name).") val output: PathPrefix,
+  @arg(flag='l', doc="Split by library instead of read group") val byLibrary: Boolean = false,
+  @arg(flag='u', doc="The name to use for the unknown file") val unknown: String = "unknown"
+) extends FgBioTool with LazyLogging {
+
+  Io.assertReadable(input)
+  Io.assertCanWriteFile(output)
+
+  override def execute(): Unit = {
+    val in       = SamReaderFactory.make().open(input)
+    val progress = ProgressLogger(logger)
+    val unknownBamAndWriter = {
+      val unknownBam: PathToBam = toOutput(unknown)
+      val unknownWriter: SAMFileWriter = new SAMFileWriterFactory().makeWriter(in.getFileHeader, true, toOutput(unknown).toFile, null)
+      WriterInfo(name=unknown, bam=unknownBam, writer=unknownWriter)
+    }
+    val writers  = createWriters(header=in.getFileHeader, unknownBamAndWriter=unknownBamAndWriter)
```

Comment:
I see .... you're just using it as the default in the map.  I think in that case it would be clearer if you just changed this line to:
`createWriters(header=in.getFileHeader).withDefaultValue(unknownBamAndWriter)`